source/use-cases/sensitive-data-encryption.txt

```diff
@@ -40,9 +40,42 @@ Procedure
 Requirements
 ~~~~~~~~~~~~
 
-- Point to setup and install directions
-- Nullam imperdiet lorem vitae vulputate lacinia.
-- Donec eget velit tincidunt, gravida diam ac, efficitur lacus.
+MongoDB Server 4.2 Enterprise
+  * For installation instructions, refer to the 
+    `Enterprise Edition Installation Tutorials <https://docs.mongodb.com/manual/installation/#mongodb-enterprise-edition-installation-tutorials>`_.
+
+
+MongoDB Driver Compatible with CSFLE 
+  * For a list of drivers that support CSFLE, refer to the
+    `driver compatibility table <https://docs.mongodb.com/manual/core/security-client-side-encryption/#driver-compatibility-table>`_.
+
+
+AWS Key Management Service (KMS)
+  * AWS KMS is required for the secure deployment of CSFLE.
+
+  * Compatible drivers currently include integration with AWS KMS. Support for
+    other KMS providers may be added in the future.
+
+  * The client application will need AWS KMS permissions to the GenerateDataKey,
+    Encrypt, and Decrypt actions.
+ 
+
+MongoDB Authorization Privileges
+  * In order to create data encryption keys, the client needs the
+    `ReadWrite <https://docs.mongodb.com/manual/reference/built-in-roles/#readWrite>`_ 
+    built-in role on the key vault collection which is specified in the client 
+    configuration.
+
+  * If no additional data encryption keys need to be created, the client can 
+    be downgraded to the `listCollections <https://docs.mongodb.com/manual/reference/privilege-actions/#listCollections>`_
+    privilege on the key vault collection.
```

Comment:
nit: Let's not frame this as "downgraded" since they likely wouldn't ever have the elevated privileges in the first place

Reply:
I was trying to identify the difference between what they permissions they would need for the data key creation step and everything following that.

On second thought, this auth requirement might be trivial to even include in this guide, especially if we assume the user is running through the guide using a local MongoDB replica set in which there are no auth restrictions.

Do you think it would be better to move this requirement into one of the steps (e.g. "Make sure your user has ReadWrite permissions for this Data Key setup step") or remove it?

Reply:
We definitely should include the required permissions since I presume enterprise users will deploy to production at some point which won't use localhost. We should also include it in the steps as an admonition.

A few things about the mongodb roles:

- Can we reframe the bullet points to be more independent?
   - e.g.
     ```rst
     - To create data encryption keys...
     - To encrypt and decrypt data...
     ```
- These bullets seem inconsistent:
   - `readWrite` is a built-in database user role
   - `listCollections` is a privilege/action granted by both the `read` and `readWrite` built-in database user roles
  - Built-in roles apply to entire databases, so you would have a built-in role _on the database that contains the keyvault collection._

Reply:
Discussed in-person; will remove this section and create a note in the Data Encryption Key section that explains what permissions are needed to create the key only. FLE team still waffling on where best to keep keys secure and standard naming across examples.